src/utils/upload/processUpload.js

```diff
@@ -209,17 +209,28 @@ const processUpload = async (filesList, sampleType, samples, activeProjectUuid,
   }, {});
 
   Object.entries(samplesMap).forEach(async ([name, sample]) => {
-    // Create sample if not exists
-    sample.uuid ??= await dispatch(createSample(activeProjectUuid, name, sampleType));
+    // Create sample if not exists.
+    try {
+      sample.uuid ??= await dispatch(createSample(activeProjectUuid, name, sampleType));
+    } catch (e) {
+      // If sample creation fails, sample should not be created
```

Comment:
log the error

Reply:
you will want to return the error to the user, no?

Reply:
by logging the error, I meant for us -- we want to see it in the console when it happens for debugging

Reply:
Ah ok, I'll add the log inside `createSample`

Reply:
Oh we already log it in these lines inside `createSample` : https://github.com/hms-dbmi-cellenics/ui/blob/7127b0dc9dcc9684807b63c61d6d4c818d55146e/src/redux/actions/samples/createSample.js#L72-L76

However, it logs the error only if it's not a server error. I don't know why this was implemented... maybe because if it's a server error (e.g. API error), we'd already have a log of it and not show it in the browser? Should we log this anyway?

Reply:
can you not just do console.error(e) in this current piece of code? When debugging, we want to see the actual error

Reply:
Ok, I'll just remove the if statement so we log all errors